### PR TITLE
RTEMIS-0: Invalidating the cache tags of the home page when updating the entity queue.

### PR DESCRIPTION
**Pull request by kunal-sin**
Ticket # : N/A

---

### CODE REVIEW

File: rte_mis.profile [hunk 1]
```diff
@@ -15,3 +16,16 @@ function rte_mis_install_tasks_alter(array &$tasks) {
   // site settings.
   $tasks['install_configure_form']['function'] = SiteConfigureForm::class;
 }
+
+/**
+ * Implements hook_update_N().
+ */
+function rte_mis_update_10006() {
```
Comment (manish-3108) [inline rte_mis.profile:23]: I think this should be added in `rte_mis.install` file.

Cc @kunal-sin

Comment (kumareshinnoraft) [reply rte_mis.profile]: @manish-3108 Updated the PR. @kunal-sin sir, please review.

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    ModuleHandlerInterface $module_handler,
+    LoggerChannelFactoryInterface $logger_factory,
+    FileSystemInterface $file_system,
+  ) {
+    $this->configFactory = $config_factory;
+    $this->languageManager = $language_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->themeManager = $theme_manager;
+    $this->moduleHandler = $module_handler;
+    $this->logger = $logger_factory->get('rte_mis_config');
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:143]: Is this valid?

Comment (kunal-sin) [reply modules/rte_mis_core/src/Helper/ConfigManager.php:143]: I think this should be tagged to `rte_mis_core`

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+    if (empty($configs)) {
+      return;
+    }
+
+    if (!in_array($path, ['install', 'optional'])) {
+      $this->logger->error('If you are trying to apply the overrides, please resave (MODE_RESAVE) original config.');
+      throw new \InvalidArgumentException('Only original config should be updated.');
+    }
+
+    // Skip updating configs for modules currently not installed.
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:171]: Please add a proper comment here.

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+   */
+  public function updateConfigs(array $configs, $module_name, $path = 'install', $mode = self::MODE_ADD_MISSING, array $options = []) {
+    if (empty($configs)) {
+      return;
+    }
+
+    if (!in_array($path, ['install', 'optional'])) {
+      $this->logger->error('If you are trying to apply the overrides, please resave (MODE_RESAVE) original config.');
+      throw new \InvalidArgumentException('Only original config should be updated.');
+    }
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:169]: Please update the logger message here with a proper one. Like only `install` `optional` configuration will be considered.

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+
+    // Skip updating configs for modules currently not installed.
+    if (!($this->moduleHandler->moduleExists($module_name))) {
+      return;
+    }
+
+    foreach ($configs as $config_id) {
+      // Be nice to devs, forgive them if they add .yml in config name.
+      $config_id = str_replace('.yml', '', $config_id);
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:178]: This is not required. Pls remove this.

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+          $index->save();
+        }
+        catch (\Throwable $e) {
+          watchdog_exception('rte_mis_config', $e);
+        }
+
+        $this->logger->info('Re-saved index @index as config saved.', [
+          '@index' => $index_name,
+        ]);
+      }
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:292]: I don't think we are using search in RTE, Please remove this.

File: modules/rte_mis_core/src/Helper/ConfigManager.php [hunk 7]
```diff
@@ -0,0 +1,519 @@
...
+          foreach ($this->moduleHandler->getModuleList() as $module) {
+            $module_path = $this->fileSystem->realpath('modules/' . $module->getName());
+            $override_path = $module_path . '/config/replace/' . $options['config_name'] . '.yml';
+
+            // If there is a replace, we replace the complete configuration.
+            if (file_exists($override_path)) {
+              $data = Yaml::parse(file_get_contents($override_path));
+            }
+          }
+        }
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Helper/ConfigManager.php:372]: @RohanSingh-inno Can we try this? For file name update? Placing the configuration inside replace folder.

**Pull request merged**